### ISSUE TITLE
[vscode web] test live share

### ISSUE THREAD

**Issue by profy12** [O-clock-Dev/teleporter-compose#98]
We should test if vscode liveshare can be activated on the web version.

**Comment by clementgoclock** [comment]
[Codespaces](https://github.com/codespaces) (vscode in browser from github)
![image](https://github.com/O-clock-Dev/teleporter-compose/assets/126659374/6d508eca-6300-4197-af3f-d64cc38623b0)
Our vscode-server
![image](https://github.com/O-clock-Dev/teleporter-compose/assets/126659374/4cb35e75-7ea9-4aeb-a4bd-2c69b24abf10)

Don't know why it's available on the first one and not the second. Though it could be devcontainer but extension wasn't installed on first one. So.. vscode-server image ?

Anyway if it's available in CodeSpace i guess we can do something here..

**Comment by clementgoclock** [comment]
Codespace
![image](https://github.com/O-clock-Dev/teleporter-compose/assets/126659374/4f49e992-1cf2-42be-814e-1ebd786a6cd1)
code-server
![image](https://github.com/O-clock-Dev/teleporter-compose/assets/126659374/ea536313-bec2-4bdd-b073-4b0081427da2)

**Comment by clementgoclock** [comment]
Could be something related to bugs/missconfiguration on our side, eg.:
```
config/vscode/.local/share/code-server/logs/20240530T173306/remoteCLI.log:1:2024-05-30 17:33:07.052 [error] Error: Failed Installing Extensions: ms-vsliveshare.vsliveshare
.env:30:VSCODE_EXTENSIONS_LIST=aaron-bond.better-comments|bierner.markdown-preview-github-styles|bmewburn.vscode-intelephense-client|dbaeumer.vscode-eslint|DavidAnson.vscode-markdownlint|dracula-theme.theme-dracula|EditorConfig.EditorConfig|file-icons.file-icons|MS-vsliveshare.vsliveshare|MS-CEINTL.vscode-language-pack-fr|ph-hawkins.arc-plus|stylelint.vscode-stylelint|wayou.vscode-todo-highlight|whatwedo.twig|yzhang.markdown-all-in-one
```
But no extensions installed.

**Comment by clementgoclock** [comment]
Smells bad https://coder.com/docs/code-server/FAQ#why-cant-code-server-use-microsofts-extension-marketplace

Web alternative:
vscode.dev
github.dev

=> supports liveshare but not devcontainer and not local. Couldnt find officiel MS VSCode web :/

Not sure liveshare is open source extension as well. Looks like we're ducked here.


Alternative is to run a real vscode and stream it through webrtc ... :D 

![image](https://github.com/O-clock-Dev/teleporter-compose/assets/126659374/7228ada7-ec9b-4685-97be-b3bb32807f38)

**Comment by profy12** [comment]
Have you tried with /var/run/docker.sock mounted (dind mode), then use devcontainers with the teleporter-terminal and then test liveshare ?

**Comment by clementgoclock** [comment]
DevContainers & LiveShare extension are closed source (tell me if i'm wrong) and Microsoft forbid vscode fork to access marketplace)

So alternative outside TP is to use desktop classsic vscode app and inside TP use neko to stream Vscode ([example here, old one](https://github.com/m1k1o/neko-apps/tree/master/vscode))

**Comment by profy12** [comment]
Yes we will remove vscode web from the project, when integration with devcontainers is ready.
